R/functions_io.R

```diff
@@ -464,3 +464,138 @@ ParsePlateInformation = function(cell_names, pattern='^(\\S+)_(\\d+)_([A-Z])(\\d
   
   return(plate_information)
 }
+
+#' Exports a Seurat object for visualisation with the cerebroApp (https://github.com/romanhaa/Cerebro).
+#' 
+#' @param sc A Seurat object.
+#' @param path File path for export.
+#' @param param Named list with parameters passed to the scrnaseq script.
+#' @param organism Species. Can be 'Hg', 'Mm', ... .
+#' @param assay Assay to export ('RNA').
+#' @param column_sample Metadata column containing the sample name ('orig.ident').
+#' @param column_cluster Metadata column containing the cluster identity ('seurat_clusters').
+#' @param column_ccphase Metadata column containing the cell cycle phase ('Phase').
+#' @param gene_lists Gene lists to include (NULL).
+#' @param marker_genes Marker genes table obtained from Seurat::FindMarkers (NULL).
+#' @param enriched_pathways List with enriched pathways results.
+#' @return TRUE if export was successful otherwise FALSE.
+ExportToCerebro = function(sc, path, param, project="scrnaseq", species, assay="RNA", column_sample="orig.ident", column_cluster="seurat_clusters", gene_lists = NULL, marker_genes=NULL, enriched_pathways=NULL, column_ccphase="Phase") {
+  
+  if (!(column_ccphase %in% colnames(sc[[]]))) column_ccphase = NULL
+  
+  
+  # Save the Seurat object in the cerebroApp format
+  cerebroApp::exportFromSeurat(sc,
+                               file=path,
+                               assay=assay,
+                               experiment_name=project,
+                               organism=species,
+                               column_sample=column_sample,
+                               column_cluster=column_cluster,
+                               column_nUMI=paste("nCount",assay,sep="_"),
+                               column_nGene=paste("nFeature",assay,sep="_"),
+                               column_cell_cycle_seurat = column_ccphase,
+                               add_all_meta_data=T)
+  
+  # Export function does not include all information - need to add manually
+  crb_obj = readRDS(path)
+  
+  # Add basic parameters (are hardcoded unfortunately)
```

Comment:
Why are they hard-coded? And why would you set some of the parameters below to NA if for real they weren't NA?

Reply:
I have no idea - I simply construct the object so that it can be read by cerebro. Also I cannot add other parameters, they just will not be shown (=hardcoded).

The NA values are because cerebro only expects one value but not as in our case a value for each dataset. I do not know yet how to solve this.